apps/backend/src/modules/users/user.repository.ts

```diff
@@ -41,8 +48,16 @@ class UserRepository implements Repository {
 		return user ? UserEntity.initialize(user) : null;
 	}
 
-	public async findAll(): Promise<UserEntity[]> {
-		const users = await this.userModel.query().execute();
+	public async findAll({
+		page,
+		pageSize,
+	}: PaginationParameters): Promise<UserEntity[]> {
+		const offset = --page * pageSize;
+		const users = await this.userModel
```

Comment:
As I remember there is a `.page(page, count)` method in objection query builder. And it returns results and total which we need to return from the repository. Remove count method above

Reply:
Okay, I've implemented these suggested changes, so to take advantage of getting items page and their total count at the same time, now the base repository's `findAll` returns `Promise<{ items: T[] }>` (not `Promise<PaginationResponseDto<T>>` because not every item collection should be pageable) and there are errors in other repositories now, because their `findAll` are different, they return arrays of Entities. Should I change it? Some logic from the service will be moved to the repo.

Reply:
Let's return an object with items property for all findAll methods to make a consistent response type

Reply:
I think there is no need for a separate type because it may differ too much, on repository / service for now we will use either `{ items: T }` or wrap resource type into Pagination type or use ResourceGetAllResponseDto. For frontend, I think we won't need `{ items: T }`, because we will keep in the store only items as it was previously. So the only thing we will need to change on frontend is that instead of setting payload to the store we will set payload.items

Reply:
I'm sorry, I just got a bit confused, and when I started working on it, I've figured out that there are already types for findAll declared, so the only backend needs to be changed a bit. Thank you!